src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Cmdlets/GetAzureRmVMGuestPolicyReport.cs

```diff
@@ -0,0 +1,180 @@
+                            WriteObject(gcPolicyAssignmentReports, true);
+                        }                        
+                    }
+                    else
+                    {
+                        gcPolicyAssignments = GetAllGuestConfigurationAssignmentsByInitiativeName(ResourceGroupName, VMName, InitiativeName);
+                        if (gcPolicyAssignments == null || gcPolicyAssignments.Count() == 0)
+                        {
+                            var message = string.Format(UserMessages.NoGuestConfigurationPolicyReportsFoundInitiativeName, ResourceGroupName, VMName, InitiativeName);
+                            WriteObject(message);
+                        }
+                        else
+                        {
+                            WriteObject(gcPolicyAssignments, true);
+                        }
+                    }
+                    break;
+
+                case ParameterSetNames.InitiativeIdScope:
+                    if (Detailed.IsPresent)
+                    {
+                        gcPolicyAssignmentReports = GetAllGuestConfigurationAssignmentReportsByInitiativeId(ResourceGroupName, VMName, InitiativeId, Latest.IsPresent);
+
+                        if (gcPolicyAssignmentReports == null || gcPolicyAssignmentReports.Count() == 0)
+                        {
+                            var message =  string.Format(UserMessages.NoDetailedGuestConfigurationPolicyReportsFoundInitiativeId, ResourceGroupName, VMName, InitiativeId);
+                            WriteObject(message);
+                        }
+                        else
+                        {
+                            WriteObject(gcPolicyAssignmentReports, true);
+                        }                   
+                    }
+                    else
+                    {
+                        gcPolicyAssignments = GetAllGuestConfigurationAssignmentsByInitiativeId(ResourceGroupName, VMName, InitiativeId);
+
+                        if (gcPolicyAssignments == null || gcPolicyAssignments.Count() == 0)
+                        {
+                            var message = string.Format(UserMessages.NoGuestConfigurationPolicyReportsFoundInitiativeId, ResourceGroupName, VMName, InitiativeId);
+                            WriteObject(message);
+                        }
+                        else
+                        {
+                            WriteObject(gcPolicyAssignments, true);
+                        }                       
+                    }
+                    break;
+
+                case ParameterSetNames.ReportIdScope:
+                    var policyReport = GetGuestConfigurationAssignmentReportById(ReportId);
+                    if (policyReport != null)
+                    {
+                        WriteObject(policyReport);
+                    }
+                    else {
+                        WriteObject(string.Format(UserMessages.InvalidReportId, ReportId));
```

Comment:
In thei case, you would return nothing.  You could also write an error or warning, but you should not write strings to the output stream